network-structure.md

```diff
@@ -0,0 +1,29 @@
+# Network structure
+
+## Node
+A Node is a local context in which (part of) an ECP deployment runs. 
+This may be a single application using one or more threads or processes. 
+An ECP network has one or more Nodes. 
+If it has a single Node, its Components may use the Local Message Transport (LMT/"local mode"), see below. 
+If it has multiple Nodes, they must use the Distributed Message Transport (DMT/"distributed mode").
+
+(TBC) Optional feature "Bridging Coordinator": In a DMT network, if the Components of a Node use local mode, only a Coordinator in that Node may use DMT to bridge messages to/from other Nodes. Put differently, Components in local mode may only communicate to outside their Node via a Coordinator in their Node.
+:::{admonition} TODO
+Decide if we include the Bridging Coordinator concept (I think we will).
+:::
+:::{admonition} TODO
+Decide if the "mode" is the same for all Components in a Node, and if yes, how we track which Node a Component belongs to, and how the mode information propagates within.
+:::
+
+## Message Layer
+The Message Layer is the communication layer that concerns itself with ECP message (de)composition, validation, serialisation, etc.
+
+## Transport Layer
+The Transport Layer is the communication layer that transports ECP messages between Components.
+This uses zeromq or simpler localised methods, see the next section "Message Transport Mode".
```

Comment:
I'm unsure, whether we understand the same things. In my opinion:

Transport Layer is not only zeromq (or other), but also contains the routing information in one node or between nodes. So #22 is part of the transport layer.

Do you understand "message" as the transported payload (then I agree with your definitions) or as the whole communicated data (including addresses etc.)?

I'd define, for clarification:
```suggestion
## ECP Message
An ECP Message is one set of data transmitted from one Component to another one. It consists of a Header, necessary to deliver the Message to its destination, and of some Content transmitted.

## Message Layer
The Message Layer is the communication layer that concerns itself with (de)composition, validation, serialisation, etc. of ECP Message Contents.

## Transport Layer
The Transport Layer is the communication layer that transports ECP messages between Components.
This uses zeromq or simpler localised methods, see the next section "Message Transport Mode".
```

Reply:
> Transport Layer is not only zeromq (or other), but also contains the routing information in one node or between nodes. So https://github.com/pymeasure/experiment_control_protocol/issues/22 is part of the transport layer.

Good point!

> Do you understand "message" as the transported payload (then I agree with your definitions) or as the whole communicated data (including addresses etc.)?

The whole data, header, payload, checksum.

Reply:
I pushed an update.